documentation/cadl-structure-guidelines.md

```diff
@@ -1,17 +1,34 @@
-# Repository Guidelines for Cadl Specs
-
-## Purpose
+<!-- vscode-markdown-toc -->
```

Comment:
We should update this doc and add pointers to the new docs being worked on at https://github.com/Azure/azure-rest-api-specs/pull/21722. We should also make sure those docs match these changes in this PR.